tests/dynamic_data/publications/layer_geoserver_proxy/geoserver_proxy_test.py

```diff
@@ -0,0 +1,140 @@
+from tests.dynamic_data import base_test
+from tests.dynamic_data.publications import common_publications as publications
+from test_tools import process_client, wfs_client
+
+pytest_generate_tests = base_test.pytest_generate_tests
+WORKSPACE = 'geoserver_proxy_wfs_test_ws'
+USER = 'geoserver_proxy_wfs_authz_test_user'
+USER_2 = 'geoserver_proxy_wfs_authz_test_user_2'
+
+PUBLIC_LAYER = Publication4Test(WORKSPACE, process_client.LAYER_TYPE, 'geoserver_proxy_wfs_test_public_layer', uuid='cab68845-7f33-44a2-8696-c55b7ca03393')
+PRIVATE_LAYER = Publication4Test(USER, process_client.LAYER_TYPE, 'geoserver_proxy_wfs_test_private_layer', uuid='a97b78b9-f92b-4ed9-9eac-fea775f1bf48')
+
+OK_TEST_CASES = {
+    'public_layer_specific_url': {
+        'layer': PUBLIC_LAYER,
+        'wfs_params': {
+            'request_workspace': GEOSERVER_WFS_WORKSPACE,
+        },
+    },
+    'public_layer_generic_url': {
+        'layer': PUBLIC_LAYER,
+        'wfs_params': {
+        },
+    },
+    'private_layer_specific_url': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_workspace': GEOSERVER_WFS_WORKSPACE,
+            'request_headers': process_client.get_authz_headers(username=USER),
+        },
+    },
+    'private_layer_generic_url': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_headers': process_client.get_authz_headers(username=USER),
+        },
+    },
+}
+
+ERROR_TEST_CASES = {
+    'private_layer_specific_url_unauthz_user': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_workspace': GEOSERVER_WFS_WORKSPACE,
+            'request_headers': process_client.get_authz_headers(username=USER_2),
+        },
+    },
+    'private_layer_generic_url_unauthz_user': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_headers': process_client.get_authz_headers(username=USER_2),
+        },
+    },
+    'private_layer_specific_url_no_user': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_workspace': GEOSERVER_WFS_WORKSPACE,
+        },
+    },
+    'private_layer_generic_url_no_user': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+        },
+    },
+    'private_layer_generic_url_fraud_header': {
+        'layer': PRIVATE_LAYER,
+        'wfs_params': {
+            'request_headers': {settings.LAYMAN_GS_AUTHN_HTTP_HEADER_ATTRIBUTE: USER, },
+        },
+    },
+}
+
+
+@pytest.mark.usefixtures('oauth2_provider_mock')
+class TestWfsProxyOkCall(base_test.TestSingleRestPublication):
+    workspace = WORKSPACE
+    publication_type = process_client.LAYER_TYPE
+    rest_parametrization = []
+    usernames_to_reserve = [USER, USER_2]
+
+    test_cases = [
+        base_test.TestCaseType(
+            key=key,
+            publication=params['layer'],
+            type=EnumTestTypes.MANDATORY,
+            params=copy.deepcopy(params['wfs_params']),
```

Comment:
I prefer to copy the whole `params` instead of `params['wfs_params']` because of readability in test method

Reply:
Changed.